lib/rspec/core/configuration.rb

```diff
@@ -1613,6 +1613,9 @@ def extract_location(path)
           captures = match.captures
           path, lines = captures[0], captures[1][1..-1].split(":").map { |n| n.to_i }
           filter_manager.add_location path, lines
+        else
+          path, scoped_ids = path.split(/[\[\]]/)
```

Comment:
> perhaps the regex needs a name? `split(match_square_brackets)` not convinced that's better, just throwing the idea out?

I'm not convinced it needs a name but am happy to assign it to a local if you find that easier to read.  I could name the local `on_square_brackets` and then it would read `split(on_square_brackets)` -- do you like that?